src/Core/Domain/Product/ValueObject/ProductRedirectOption.php

```diff
@@ -0,0 +1,227 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\ValueObject;
+
+use LogicException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductConstraintException;
+use PrestaShop\PrestaShop\Core\Product\ProductInterface;
+
+/**
+ * Holds values for product redirect option.
+ * todo: there are type constants declared already in @var ProductInterface. Should I reuse them? deprecate them?
```

Comment:
I dont know 😅  i'm fifty-fifty on this, between "reuse it" or "deprecate it"

@PrestaShop/prestashop-core-developers help 😅

Reply:
still relevant

Reply:
I would be in favor to keep a single place to store these, maybe not necessarily in `ProductInterface` if you think it's not the appropriate place
But I don't think these should be define in a ValueObject either, usually in this kind of situation we create a dedicated class like `PrestaShop\PrestaShop\Core\Domain\Product\ProductRedirectionSettings` (so ProductInterface would use it to populate its const and could be deprecated)

Reply:
> But I don't think these should be define in a ValueObject either, usually in this kind of situation we create a dedicated class like PrestaShop\PrestaShop\Core\Domain\Product\ProductRedirectionSettings (so ProductInterface would use it to populate its const and could be deprecated)

Agreed 👍